File: trunk/base/doc/guides/PrimaryProducerExample.c
#include <stdio.h>
#include <stdlib.h>

#include "rgma/rgma.h"

#define MAXSTR 255

main(int argc, char *argv[]) {
    RGMAPrimaryProducer *pp;
    char insert[MAXSTR], predicate[MAXSTR];
    RGMAException * exception;

    if (argc != 2) {
        fprintf(stderr, "Usage: %s <userId>\n", argv[0]);
        exit(1);
    }

    pp = RGMAPrimaryProducer_create(RGMAStorageType_MEMORY, NULL,
            RGMASupportedQueries_C, &exception);
    if (exception) {
        fprintf(stderr, "Failed to create producer %s\n", exception->message);
        exit(1);
    }

    snprintf(predicate, MAXSTR, "WHERE userId = '%s'", argv[1]);

    RGMAPrimaryProducer_declareTable(pp, "default.userTable", predicate,
            3600, 3600, &exception);
    if (exception) {
        fprintf(stderr, "Failed to declare table %s\n", exception->message);
        exit(1);
    }

    snprintf(insert, MAXSTR, "INSERT INTO default.userTable (userId, aString, "
            "aReal, anInt) VALUES ('%s', 'C producer', 3.1415926, 42)", argv[1]);

    RGMAPrimaryProducer_insert(pp, insert, 0, &exception);
    if (exception) {
        fprintf(stderr, "Failed to insert %s\n", exception->message);
        exit(1);
    }

    RGMAPrimaryProducer_close(pp, &exception);
    if (exception) {
        fprintf(stderr, "Failed to close %s\n", exception->message);
        exit(1);
    }

    exit(0);
}

File: base/doc/guides/PrimaryProducerExample.py
#!/usr/bin/env python

import sys
import rgma

if len(sys.argv) != 2:
    sys.stderr.write("Exactly one argument must be specified\n")
    sys.exit(1)

userId = sys.argv[1]

try:
    producer = rgma.PrimaryProducer(rgma.Storage.getMemoryStorage(), 
                                    rgma.SupportedQueries.C)

    predicate =  "WHERE userId = '%s'" % userId
    historyRetentionPeriod = rgma.TimeInterval(60, rgma.TimeUnit.MINUTES)
    latestRetentionPeriod = rgma.TimeInterval(60, rgma.TimeUnit.MINUTES)
    producer.declareTable("default.userTable", predicate,
                          latestRetentionPeriod, historyRetentionPeriod)

    insert = "INSERT INTO default.userTable " \
             "(userId, aString, aReal, anInt) " \
             "VALUES ('%s', 'Python producer', 3.1415962, 42)" % userId
    producer.insert(insert)
    
    producer.close()

except rgma.RGMAException, e:
    sys.stderr.write("RGMAException " +  e.getMessage() + "\n")
    sys.exit(1)
